prog/weaver/main.go

```diff
@@ -506,6 +512,22 @@ func main() {
 		go exposeForAWSVPC(allocator, defaultSubnet, bridgeConfig.WeaveBridgeName, waitReady.Add())
 	}
 
+	if iptablesRefresh > 0 {
+		Log.Printf("Entering iptable refresh loop (interval %v)", iptablesRefresh)
+		applyIPTables := func() {
+			Log.Info("Re-configuring iptables")
+			err := weavenet.ConfigureIPTables(&bridgeConfig, ips)
+			if err != nil {
+				Log.Errorf("Error configuring iptables: %s", err)
+			}
+
+			weavenet.Reexpose(&bridgeConfig, Log)
+		}
+		go weavenet.Monitor(Log, "WEAVE-CANARY", []string{"mangle", "nat", "filter"}, applyIPTables, iptablesRefresh, wait.NeverStop)
```

Comment:
We do shut down other goroutines started here, generally with some kind of `defer Stop()`, so creating a `chan` and `defer close()`-ing it would be preferable to `NeverStop`.